### DISCUSSION TITLE
stories_things/basic_content/virtual/

### DISCUSSION THREAD

**Discussion by giscus** [Light-City/light-comments#11] [Announcements]
# stories_things/basic_content/virtual/

Stories About C Plus Plus

https://light-city.github.io/stories_things/basic_content/virtual/

**Comment by cxz0218** [comment]
大佬，warn_rtti.cpp中 class F : private B {} ，F是私有继承B的，那么F是不是应该为非多态类型呀~

// 被转换对象obj的类型T1必须是多态类型，即T1必须公有继承自其它类，或者T1拥有虚函数（继承或自定义）。

**Reply by Light-City** [reply]
T1拥有虚函数（继承或自定义） 也可以
